cmake/SetupCodeChecks.cmake

```diff
@@ -72,10 +72,21 @@ if(CPPCHECK_FOUND)
     add_dependencies(check cppcheck_check)
 endif()
 
+if(CLANGQUERY_FOUND)
+    # note: interactive_clang_query_check 
+    # is for the use of code developers who
+    # want to check specific attributes of
+    # specific targets, and does not make 
+    # sense as a dependency of check
+    add_custom_target(clang_query_check)
+    add_custom_target(interactive_clang_query_check)
+    add_dependencies(check clang_query_check)
+endif()
+
 # Code check targets should only be run on demand
 foreach(target 
         check uncrustify_check astyle_check cppcheck_check
-        style uncrustify_style astyle_style )
+        style uncrustify_style astyle_style clang_query_check interactive_clang_query_check)
```

Comment:
Suggestion: Perhaps it would be easier to read these if they were on their own line?

Reply:
Done